frontend/src/modules/otherlocales/components/OtherLocales.js

```diff
@@ -37,41 +35,56 @@ export default class OtherLocales extends React.Component<Props> {
         </section>
     }
 
+    renderTranslations(
+        translation: OtherLocaleTranslation,
+        index: number,
+    ) {
+        return <Translation
+            entity={ this.props.entity }
+            isReadOnlyEditor={ this.props.isReadOnlyEditor }
+            translation={ translation }
+            parameters={ this.props.parameters }
+            updateEditorTranslation={ this.props.updateEditorTranslation }
+            key={ index }
+        />;
+    }
+
     render() {
-        const {
-            entity,
-            isReadOnlyEditor,
-            orderedOtherLocales,
-            preferredLocalesCount,
-            otherlocales,
-            parameters,
-            updateEditorTranslation,
-        } = this.props;
+        const { otherlocales } = this.props;
+
+        if (!otherlocales.translations) {
+            return null;
+        }
+
+        const translation = otherlocales.translations;
 
         if (otherlocales.fetching) {
             return null;
         }
 
-        if (!otherlocales.translations.length) {
+        if (!translation.other.length && !translation.preferred.length) {
             return this.renderNoResults();
         }
 
         return <section className="other-locales">
-            <ul>
-                { orderedOtherLocales.map((translation, index) => {
-                    let lastPreferred = (index === preferredLocalesCount - 1);
-
-                    return <Translation
-                        entity={ entity }
-                        isReadOnlyEditor={ isReadOnlyEditor }
-                        translation={ translation }
-                        parameters={ parameters }
-                        updateEditorTranslation={ updateEditorTranslation }
-                        lastPreferred={ lastPreferred }
-                        key={ index }
-                    />;
-                }) }
-            </ul>
-        </section>;
+                <ul className="preferred-list">
+                    { translation.preferred.map((translation, index) => {
+
```

Comment:
I'd remove this blank line (and the one below). And maybe add one to separate `<ul>`'s.